src/dev/run_find_plugins_with_circular_deps.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import dedent from 'dedent';
+import { parseDependencyTree, parseCircular, prettyCircular } from 'dpdm';
+import { differenceWith, fromPairs, isEqual, toPairs } from 'lodash';
+import { run } from '@kbn/dev-utils';
+
+interface Options {
+  debug?: boolean;
+}
+
+interface CircularDepList {
+  [key: string]: string;
+}
+
+const allowedList: CircularDepList = {
+  'src/plugins/data': 'src/plugins/expressions',
+  'src/plugins/expressions': 'src/plugins/visualizations',
+  'src/plugins/home': 'src/plugins/discover',
+  'src/plugins/ui_actions': 'src/plugins/embeddable',
+  'src/plugins/vis_default_editor': 'src/plugins/visualize',
+  'src/plugins/visualizations': 'src/plugins/vis_default_editor',
+  'x-pack/plugins/apm': 'x-pack/plugins/infra',
+  'x-pack/plugins/lists': 'x-pack/plugins/security_solution',
+  'x-pack/plugins/security': 'x-pack/plugins/spaces',
+};
+
+run(
+  async ({ flags, log }) => {
+    const { debug = false } = flags as Options;
+    const foundList: CircularDepList = {};
+    const depTree = await parseDependencyTree(['{src,x-pack}/plugins/**/*'], {
```

Comment:
Do we want to use [simple plugin discovery](https://github.com/elastic/kibana/blob/master/packages/kbn-dev-utils/src/plugins/simple_kibana_platform_plugin_discovery.ts) to find plugins? Do we need to include example plugins?

Reply:
We only want to search circular dependencies across plugins we build/maintain under the Kibana repository so I believe we don't need to use the simple plugin discovery as it will include `kibana-extra` and `plugins` folder. 

Regarding the example plugins I have include them in the script 😃

Reply:
We could also drop third party plugins from simple plugin discovery with an argument to that function, I'm not sold on it but I would prefer to not replicate the plugin discovery logic into more places than necessary. Your call though.